azurelinuxagent/common/cgroup.py

```diff
@@ -152,29 +187,41 @@ def collect(self):
 
         :rtype: [(str, str, float)]
         """
-        self.update()
-        usage = self.get_cpu_percent()
-        return [("Process", "% Processor Time", usage)]
+        self._update_cpu_data()
+        usage = self._get_cpu_percent()
+        return [CollectedMetrics("cpu", "% Processor Time", usage)]
 
 
-class MemoryCGroup(CGroup):
-    def __init__(self, cgt):
+class MemoryCgroup(CGroup):
+    def __init__(self, extension_name, cgroup_path, controller="memory"):
         """
         Initialize data collection for the Memory hierarchy
 
         :param CGroupsTelemetry cgt: The telemetry object for which memory metrics should be collected
         :return:
         """
-        self.cgt = cgt
+        super(MemoryCgroup, self).__init__(extension_name, cgroup_path, controller)
+
+    def _get_memory_usage(self):
+        """
+        Collect memory.usage_in_bytes from the cgroup.
+
+        :return: Memory usage in bytes
+        :rtype: int
+        """
+        usage = self.get_parameter('memory', 'memory.usage_in_bytes')
```

Comment:
Is there a reason you are not using `self.controller` instead of `'memory'` like in L224?

Reply:
Removed the controller itself. No need to handle this hot-potato. :)